sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Filtering/Filter.cs

```diff
@@ -0,0 +1,719 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Globalization;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Reflection;
+    using Azure.Monitor.OpenTelemetry.LiveMetrics.Models;
+
+    /// <summary>
+    /// Filter determines whether a telemetry document matches the criterion.
+    /// The filter's configuration (condition) is specified in a <see cref="FilterInfo"/> DTO.
+    /// </summary>
+    /// <typeparam name="TTelemetry">Type of telemetry documents.</typeparam>
+    internal class Filter<TTelemetry>
+    {
+        private const string FieldNameCustomDimensionsPrefix = "CustomDimensions.";
+
+        private const string FieldNameCustomMetricsPrefix = "CustomMetrics.";
+
+        private const string FieldNameAsterisk = "*";
+
+        private const string CustomMetricsPropertyName = "Metrics";
+
+        private const string CustomDimensionsPropertyName = "Properties";
+
+        private const char FieldNameTrainSeparator = '.';
+
+        private static readonly MethodInfo DoubleToStringMethodInfo = GetMethodInfo<double, string>(x => x.ToString(CultureInfo.InvariantCulture));
+
+        private static readonly MethodInfo NullableDoubleToStringMethodInfo = GetMethodInfo<double?, string>(x => x.ToString());
+
+        private static readonly MethodInfo ObjectToStringMethodInfo = GetMethodInfo<object, string>(x => x.ToString());
+
+        private static readonly MethodInfo ValueTypeToStringMethodInfo = GetMethodInfo<ValueType, string>(x => x.ToString());
+
+        private static readonly MethodInfo UriToStringMethodInfo = GetMethodInfo<Uri, string>(x => x.AbsoluteUri);
```

Comment:
Why this variable has changed from AI SDK?

Reply:
It was to resolve a compiler error. I'll revert it as it's also causing an issue in my next PR.